src/System.Diagnostics.PerformanceCounter/tests/PerformanceDataTests.cs

```diff
@@ -0,0 +1,251 @@
+                using (CounterSetInstance typingCsInstance = typingCounterSet.CreateCounterSetInstance("Typing Instance"))
+                {
+                    Assert.Throws<ArgumentException>("instanceName", () => typingCounterSet.CreateCounterSetInstance("Typing Instance"));
+                }
+            }
+        }
+
+        [ConditionalFact(typeof(Helpers), nameof(Helpers.IsElevatedAndCanWriteToPerfCounters))]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+        public void PerformanceCounter_PerformanceData_CounterSet_AlreadyRegistered()
+        {
+            using (CounterSet typingCounterSet = new CounterSet(_fixture._providerId, _fixture._typingCounterSetId, CounterSetInstanceType.Single))
+            {
+                Assert.Throws<ArgumentException>("counterSetGuid", () => new CounterSet(_fixture._providerId, _fixture._typingCounterSetId, CounterSetInstanceType.Single));
+            }
+        }
+
+        [ConditionalTheory(typeof(Helpers), nameof(Helpers.IsElevatedAndCanWriteToPerfCounters))]
+        [InlineData("", typeof(ArgumentException))]
+        [InlineData(null, typeof(ArgumentNullException))]
+        public void PerformanceCounter_PerformanceData_CounterSet_InvalidInstanceName(string instanceName, Type exceptionType)
+        {
+            using (CounterSet typingCounterSet = new CounterSet(_fixture._providerId, _fixture._typingCounterSetId, CounterSetInstanceType.Single))
+            {
+                typingCounterSet.AddCounter(6, CounterType.SampleBase, "Percent Base");
+                ArgumentException argumentException = (ArgumentException)Assert.Throws(exceptionType, () => typingCounterSet.CreateCounterSetInstance(instanceName));
+                Assert.Equal("instanceName", argumentException.ParamName);
+            }
+        }
+
+        [ConditionalTheory(typeof(Helpers), nameof(Helpers.IsElevatedAndCanWriteToPerfCounters))]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+        [InlineData(8, "", "counterName", typeof(ArgumentException))]
+        [InlineData(8, null, "counterName", typeof(ArgumentNullException))]
+        public void PerformanceCounter_PerformanceData_AddCounter_InvalidCounterName(int counterId, string counterName, string parameterName, Type exceptionType)
+        {
+            using (CounterSet typingCounterSet = new CounterSet(_fixture._providerId, _fixture._typingCounterSetId, CounterSetInstanceType.Single))
+            {
+                ArgumentException argumentException = (ArgumentException)Assert.Throws(exceptionType, () => typingCounterSet.AddCounter(counterId, CounterType.SampleBase, counterName));
+                Assert.Equal(parameterName, argumentException.ParamName);
+            }
+        }
+
+        [ConditionalTheory(typeof(Helpers), nameof(Helpers.IsElevatedAndCanWriteToPerfCounters))]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
```

Comment:
I need to skip on netfx because after amend with `nameof(...)` on args exception on full framework will fail.

Reply:
There are `Assert.Throws` overloads that allow for different argument names in exceptions between netfx and netcore:

`public static void Throws<T>(string netCoreParamName, string netFxParamName, ...`

Reply:
ah i forget it you'are right!